analytics/src/main/java/com/segment/analytics/SegmentIntegration.java

```diff
@@ -379,19 +388,17 @@ void performFlush() {
         int payloadsUploaded = 0;
         Client.Connection connection = null;
         try {
+            if (useSigning) {
```

Comment:
im still thinking on if there is a better way where we can go through batch once and compute both the payload and the signing key

Reply:
discussed in slack